fsm.go

```diff
@@ -57,6 +57,10 @@ type FSM struct {
 	stateMu sync.RWMutex
 	// eventMu guards access to Event() and Transition().
 	eventMu sync.Mutex
+	// data can be used to store and load data that maybe used across events
+	data map[string]*DataValue
```

Comment:
Maybe it can be implemented using the sync.Map instead, to limit the amount of moving parts in the FSM struct. What do you think? I would also prefer the name metadata to signal that it's mostly for sparse/lightweight data.

Reply:
I considered using sync.Map initially , but I was not sure whether it is the one for our use case (https://golang.org/pkg/sync/#Map).
I can go ahead and implement it using sync.Map or use a regular map + RWMutex, if we need less moving parts.
what should i do?

Reply:
Why did you not think it would be good here?

Reply:
It was mentioned in the documentation and in the [talk](https://youtu.be/C1EtfDnsdDs) that sync.Map should be only when 
1) there are stable keys or
2) concurrent reads and writes are on disjoint keys.
So I didn't use it here.

Reply:
Ah, got it. Yes, then keep the current separate mutex. I would however prefer the names `metadata` and `metadataMu` to signal that they belong together.

Reply:
sure. I will rename them and push the changes